New Luminex Surveillance results dataset workflow

spamhurts opened the pull request:
#### Rationale
The output from the new Luminex assay is entirely different from the previous assay, with new data types and a new output format. A separate workflow has been created so we can continue to support the old Luminex system while providing support for the new one.

spamhurts committed:
Updates for the SPF results datasets

spamhurts committed:
Updates for the SPF results datasets

- removed date customization from table customizer class
- updated deprecated log helper class in ehr customizer
- removed old todo regarding IACUC PI in ehr customizer

- added tab page for luminex surveillance results in animal history at Laura P's request

- added date customization to query view customization

- removed console log from javascript trigger code

spamhurts committed:
Tweaked column widths for room and location columns to prevent wrapping

spamhurts committed:
Merge branch 'release24.7-SNAPSHOT' into 24.7_fb_SPF_Results

spamhurts committed:
updated column name

jallentxbiomed commented on snprc_ehr/resources/queries/study/SPF_Results.js:
```diff
@@ -0,0 +1,7 @@
+
+function beforeInsert(row, errors) {
+    row.objectid = row.objectid || LABKEY.Utils.generateUUID().toUpperCase();
+    // console.log("SPF_Results.js:  row.id = " + row.id + "row.objectid: " + row.objectid)
```
Do we need to leave this commented console.log here?

spamhurts replied:
I got it in one of the later commits

spamhurts replied:
Oh. I see what you mean. I can remove the comment.

spamhurts replied:
I have a bad habit of leaving them in place after I use them.

spamhurts committed:
removed debug code

jallentxbiomed commented on snprc_ehr/src/org/labkey/snprc_ehr/table/SNPRC_EHRCustomizer.java:
```diff
@@ -77,46 +78,25 @@ public void customize(TableInfo table)
-/*  //TODO: need to add investigator foreign key to multiple tables once the arc_valid_pi table has been ETL'd. tjh
-    // Are we using investigatorId or investigatorName???
-
-    private void customizeColumns(AbstractTableInfo ti)
```
Are we not going to need this method anymore? It says TODO so unsure what its purpose was

spamhurts replied:
We don't need it, as I mentioned in my commit comment. We decided to use the PI names in the IACUC data.

RameshRapa commented on snprc_ehr/resources/queries/study/SPF_Results.query.xml:
```diff
@@ -0,0 +1,22 @@
+                        <columnTitle>Bleed Date</columnTitle>
+                        <formatString>Date</formatString>
+                    </column>
+                    <column columnName="Id">
```
Hi Terry. Can i please know if this ColumnName 'Id' can convey more details instead of just Id or do we need to have the same name as it is a foreign key from study.animal Table - Id? Can we add any  "ColumnTitle" attribute to this Id ?

RameshRapa replied:
previously i gave the columnTitle in open and closed tags (<>) and it is not visible in those tags.

spamhurts replied:
Id is the name used in LabKey for the animal id. We want to keep that the same.

jallentxbiomed commented on snprc_ehr/src/org/labkey/snprc_ehr/table/SNPRC_EHRCustomizer.java:
```diff
@@ -77,46 +78,25 @@ public void customize(TableInfo table)
             doTableSpecificCustomizations((AbstractTableInfo) table);
             doCalculatedCustomizations((AbstractTableInfo) table);
             //TODO: customizeColumns((AbstractTableInfo) table);
```
Can we also remove this line if we're going to remove the method?

spamhurts replied:
Sure. I also didn't remove the date customization for the spf results bleed date. I'll do that and push it at the same time.

spamhurts committed:
- Removed TODO for customizing TableColumns
- removed sfp results date column customization (at Marty's request) to determine if the query metadata is working properly on the staging server

spamhurts committed:
Merge branch 'release24.7-SNAPSHOT' into 24.7_fb_SPF_Results

spamhurts committed:
Merge branch 'release24.7-SNAPSHOT' into 24.7_fb_SPF_Results
